nautobot_secrets_providers/providers/one_password.py

```diff
@@ -0,0 +1,110 @@
+"""1Password Secrets Provider for Nautobot."""
+
+import asyncio
+
+from django import forms
+from django.conf import settings
+from nautobot.core.forms import BootstrapMixin
+from nautobot.extras.secrets import SecretsProvider, exceptions
+
+try:
+    from onepassword.client import Client
+except ImportError:
+    Client = None
+
+from nautobot_secrets_providers import __version__
+
+__all__ = ("OnePasswordSecretsProvider",)
+
+
+async def get_secret_from_vault(vault, item, field, token, section=None):
+    """Get a secret from a 1Password vault.
+
+    Args:
+        vault (str): 1Password Vault where the secret is located.
+        item (str): 1Password Item where the secret is located.
+        field (str): 1Password secret field name.
+        token (str): 1Password Service Account token.
+        section (str, optional): 1Password Item Section for the secret. Defaults to None.
+
+    Returns:
+        (str): Value from the secret.
+    """
+    client = await Client.authenticate(
+        auth=token, integration_name="nautobot-secrets-providers", integration_version=__version__
+    )
+    reference = f"op://{vault}/{item}/{f'{section}/' if section else ''}{field}"
```

Comment:
For some reason I thought you couldn't nest f-strings inside f-strings. Good to know!

Reply:
You can, this is what Sorcery recommended. And google told me that it doesn't have a lot of use cases, but is a test case in the library.